pkg/provider/azure.go

```diff
@@ -421,7 +424,11 @@ type Cloud struct {
 	*ManagedDiskController
 	*controllerCommon
 
+	// multipleStandardLoadBalancerConfigurationsSynced make sure the `reconcileMultipleStandardLoadBalancerConfigurations`
+	// runs only once every time the cloud provide restarts.
 	multipleStandardLoadBalancerConfigurationsSynced bool
+	// nodesWithCorrectLoadBalancerByPrimaryVMSet marks nodes that are matched with load balancers by primary vmSet.
+	nodesWithCorrectLoadBalancerByPrimaryVMSet sets.Set[string]
```

Comment:
the property is a little confusing and a lock should be added to avoid race conditions

Reply:
the service reconciliation is triggered in serial, do we still need a lock to guard this?

Reply:
It may be updated by different go routines, e.g. service controller and node watcher

Reply:
please check the new commit